core/kernel/tee_ta_manager.c

```diff
@@ -854,6 +854,13 @@ static TEE_Result dump_ta_memstats(struct tee_ta_session *s,
 	if (!ts_ctx)
 		return TEE_ERROR_ITEM_NOT_FOUND;
 
+	/*
+	 * if the user TA is still being initialized,
+	 * return BAD_STATE to avoid accessing it.
+	 */
+	if (to_user_ta_ctx(ts_ctx)->uctx.is_initializing)
```

Comment:
You need to check that `ts_ctx` really is a user_ta_ctx with `is_user_ta_ctx()` first.